src/VSSTypeSanitizer.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+ * ******************************************************************************
+ * Copyright (c) 2020 Robert Bosch GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/org/documents/epl-2.0/index.php
+ *
+ *  Contributors:
+ *      Robert Bosch GmbH
+ * *****************************************************************************
+ */
+
+
+#include "VssDatabase.hpp"
+#include "ILogger.hpp"
+#include "exception.hpp"
+#include <string>
+#include <boost/algorithm/string.hpp>
+
+
+
+
+
+template<typename T>
+void checkNumTypes(jsoncons::json &meta, jsoncons::json &val )
+{
+    //Double can be larger than largest allowed VSS type ((u)int64)
+    double dval = val.as<long double>();
```

Comment:
use https://en.cppreference.com/w/cpp/string/basic_string/stol or maybe there is better function for it. I did not search so much. but what you did is unsafe. see the failing testcases :p